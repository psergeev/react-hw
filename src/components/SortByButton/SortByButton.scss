.sort-by-button {
    margin-left: 30px;
    color: #666;
    text-decoration: none;

    &.selected {
        color: #F45263;
    }
}