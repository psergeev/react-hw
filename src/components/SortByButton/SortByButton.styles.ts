const styles: any = {
    button: {
        marginLeft: '20px',
        color: '#666',
        textTransform: 'none',
        fontSize: '16px',
        padding: '0'
    },

    selected: {
        color: '#F45263'
    }
};

export default styles;
