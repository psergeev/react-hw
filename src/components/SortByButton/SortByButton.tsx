import { Button, WithStyles } from '@material-ui/core';
import * as React from 'react';

export interface Props {
    value: string;
    selected: boolean;
    handleSortByClick: (value: string) => void;
}

type PropsWithStyles = Props & WithStyles<'button' | 'selected'>;

export default React.memo(({ value, selected, handleSortByClick, classes }: PropsWithStyles) => {
    const onClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
        event.preventDefault();

        return handleSortByClick(value);
    };

    return (
        <Button
            href="#"
            onClick={onClick}
            className={`${classes.button} ${(selected ? classes.selected : '')}`}
        >
            {value}
        </Button>
    );
});
