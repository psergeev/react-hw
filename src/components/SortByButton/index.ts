import { withStyles } from '@material-ui/core';
import SortByButton from './SortByButton';
import styles from './SortByButton.styles';

export default withStyles(styles)(SortByButton);
