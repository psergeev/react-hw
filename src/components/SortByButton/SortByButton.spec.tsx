import classes from 'identity-obj-proxy';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SortByButton from './SortByButton';

test('SortByButton component works as expected', () => {
    const component = renderer.create(
        <SortByButton
            classes={classes}
            value="value"
            selected={false}
            handleSortByClick={jest.fn()}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('SortByButton component can be selected', () => {
    const component = renderer.create(
        <SortByButton
            classes={classes}
            value="value"
            selected
            handleSortByClick={jest.fn()}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
