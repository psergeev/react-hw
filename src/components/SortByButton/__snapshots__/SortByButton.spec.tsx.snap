// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortByButton component can be selected 1`] = `
<a
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 button selected"
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-2"
  >
    value
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</a>
`;

exports[`SortByButton component works as expected 1`] = `
<a
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 button "
  href="#"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <span
    className="MuiButton-label-2"
  >
    value
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</a>
`;
